src/permissions/userPermissions.manager.ts

```diff
@@ -0,0 +1,112 @@
+import { IUserPermissions, PermissionTypes } from './userPermissions.interface';
+import { UserPermissionsRepository } from './userPermissions.repository';
+import { UnauthorizedUserError, ChannelNotFoundError, UserPermissionsAlredyExistsError, OwnerPermissionsCanNotBeRemovedError } from '../utils/errors/userErrors';
+import { ChannelManager } from '../channel/channel.manager';
+
+export class UserPermissionsManager {
+
+    static async create(userPermissions: IUserPermissions, requestingUser: string) {
+        const returnedResults = await Promise.all([
+            UserPermissionsManager.isUserAdmin(requestingUser, userPermissions.channel),
+            ChannelManager.getById(userPermissions.channel),
+            UserPermissionsManager.getOne(userPermissions.user, userPermissions.channel),
+        ]);
+
+        const [isRequestingUserAdmin, channel, user] = returnedResults;
+
+        if (channel) {
+            if (!user) {
+                if (isRequestingUserAdmin || channel.user === requestingUser) {
+                    return UserPermissionsRepository.create(userPermissions);
+                }
+
+                throw new UnauthorizedUserError();
+            }
+
+            throw new UserPermissionsAlredyExistsError();
```

Comment:
`UserPermissionsAlredyExistsError` letter `a` is missing in `already` word.
Should be changed to `UserPermissionsAlreadyExistsError`